cirq-core/cirq/sim/act_on_args_container.py

```diff
@@ -77,29 +77,35 @@ def create_merged_state(self) -> TActOnArgs:
             final_args = final_args.kronecker_product(args)
         return final_args.transpose_to_qubit_order(self.qubits)
 
-    def apply_operation(
+    def _act_on_fallback_(
         self,
-        op: 'cirq.Operation',
+        action: Union['cirq.Operation', 'cirq.Gate'],
+        qubits: Sequence['cirq.Qid'],
+        allow_decompose: bool = True,
     ):
-        gate = op.gate
+        if isinstance(action, ops.Operation):
+            gate = action.gate
+        else:
+            gate = action
+
         if isinstance(gate, ops.IdentityGate):
-            return
+            return True
```

Comment:
If this method has a return type, it should be documented in the function header.

Reply:
Added to base docs, also added return type spec to all implementations